common/lib/dependabot/package_manager.rb

```diff
@@ -62,12 +81,16 @@ def deprecated?
     end
 
     # Checks if the current version is unsupported.
-    # Returns true if the version is in the unsupported_versions array; false otherwise.
+    # Returns true if the version lower then all supported versions.
     # @example
     #   package_manager.unsupported? #=> false
     sig { returns(T::Boolean) }
     def unsupported?
-      false
+      # If there is no defined supported_versions, we assume that all versions are supported
+      return false if supported_versions.empty?
+
+      # Check if the version is not supported
+      supported_versions.all? { |supported| supported > version }
     end
```

Comment:
composer and bundler are implementing this method in the same way and other ecosystems are also going to apply in same way. If there is ecosystem needs to implement unsupported in a different way then it should override that in it is package manager